tests/lib/rules/jsx-no-literals.js

```diff
@@ -127,6 +127,48 @@ ruleTester.run('jsx-no-literals', rule, {
       parser: 'babel-eslint',
       options: [{noStrings: true}]
     }, {
+      code: '<Foo bar={true} />',
+      parser: 'babel-eslint',
+      options: [{noStrings: true}]
+    }, {
+      code: '<Foo bar={false} />',
+      parser: 'babel-eslint',
+      options: [{noStrings: true}]
+    }, {
+      code: '<Foo bar={100} />',
+      parser: 'babel-eslint',
+      options: [{noStrings: true}]
+    }, {
+      code: '<Foo bar={null} />',
+      parser: 'babel-eslint',
+      options: [{noStrings: true}]
+    }, {
+      code: '<Foo bar={{}} />',
+      parser: 'babel-eslint',
```

Comment:
tl;dr: if they don't generate a syntax error with the default parser, they don't need babel-eslint :-)

Generally tho, the default parser supports everything that's actually in JS. Things that aren't in JS (or aren't yet), like class properties, require babel-eslint.